Zend/zend_vm_def.h

```diff
@@ -6431,7 +6431,8 @@ ZEND_VM_COLD_CONST_HANDLER(51, ZEND_CAST, CONST|TMP|VAR|CV, ANY, TYPE)
 					}
 				} else if (Z_OBJ_P(expr)->properties == NULL
 				 && Z_OBJ_HT_P(expr)->get_properties_for == NULL
-				 && Z_OBJ_HT_P(expr)->get_properties == zend_std_get_properties) {
+				 && Z_OBJ_HT_P(expr)->get_properties == zend_std_get_properties
+				 && (!zend_object_is_lazy_proxy(Z_OBJ_P(expr)) || !zend_lazy_object_initialized(Z_OBJ_P(expr)))) {
```

Comment:
It's not completely clear to me why initialized, lazy objects cannot go through the fast path. Can you clarify in a comment?